sql/plan/json_table.go

```diff
@@ -51,12 +51,28 @@ func (j *jsonTablePartitionIter) Next(ctx *sql.Context) (sql.Partition, error) {
 	return &jsonTablePartition{key}, nil
 }
 
+type JSONTableColOpts struct {
+	Name         string
+	Type         sql.Type
+	ForOrd       bool
+	Exists       bool
+	DefEmptyVal  sql.Expression
+	DefErrorVal  sql.Expression
+	ErrorOnError bool
+	ErrorOnEmpty bool
+}
+
+type JSONTableCol struct {
+	Path string
+	Opts *JSONTableColOpts
+	Cols []JSONTableCol
```

Comment:
(minor) Might be helpful to call this `NestedCols`, just to make it more clear on how the `nested` feature maps back to the code.